File: app/ios/CacheManagerModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(CacheManagerModule, NSObject)

RCT_EXTERN_METHOD(clearHttpCache:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: app/ios/LStream/CacheManagerModule.swift
import Foundation

@objc(CacheManagerModule)
class CacheManagerModule: NSObject {

  @objc
  func clearHttpCache(
    _ resolve: @escaping RCTPromiseResolveBlock,
    rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    URLCache.shared.removeAllCachedResponses()
    resolve(nil)
  }

  @objc
  static func requiresMainQueueSetup() -> Bool {
    return false
  }
}
